[FEATURE] yarn osd clean script should clean targets

Is your feature request related to a problem? Please describe.

Currently, we need to manually remove target directories that have outdated files that stop `yarn osd bootstrap` from running.

This most commonly happens with `Cheerio` and the type definitions in `enzyme-helpers`

Describe the solution you'd like

The current yarn osd clean script should find and clean targets across the dashboards codebase

Describe alternatives you've considered

n/a

Additional context

n/a

Thanks to @ashwin-pc  @AMoo-Miki 

One of the latests commits made to OSD updates some dependencies that require you to delete existing target folders on your dev environment and bootstrapped again before starting up. This seems to only be an issue if you are pulling the latest changes and bootstrapping OSD.
Error:
```
ERROR [bootstrap] failed:
ERROR Error: Command failed with exit code 1: /usr/share/yarn/bin/yarn.js run osd:bootstrap
      error Command failed with exit code 1.
      $ node scripts/build_ts_refs && node scripts/register_git_hook
      ERROR Error: Command failed with exit code 1: /home/anan/work/OpenSearch-Dashboards/node_modules/typescript/bin/tsc -b tsconfig.refs.json --pretty
            src/test_utils/target/public/enzyme_helpers.d.ts:61:5 - error TS2304: Cannot find name 'Cheerio'.

            61 }): Cheerio;
                   ~~~~~~~


            Found 1 error. 
```

`osd clean` does not delete all the target folders and this seems to be the issue. Cleaning all the target folders using ```find . -name 'target' -type d -prune -exec rm -rf '{}' + ``` (Credits [@miki] and then running osd clean should allow the bootstrap to work

--- a/package.json
+++ b/package.json
@@ -31,7 +31,9 @@
         "built_assets",
         ".eslintcache",
         ".node_binaries",
-        "src/plugins/*/target"
+        "src/plugins/*/target",
+        "src/core/target",
+        "src/test_utils/target"
       ]
     }
   },

